exporter/splunkhecexporter/config.go

```diff
@@ -46,6 +47,13 @@ type OtelToHecFields struct {
 	SeverityNumber string `mapstructure:"severity_number"`
 }
 
+type WorkerQueueConfig struct {
```

Comment:
This should be out of this PR for now so we can have a PR just for this, and I don't think this is something we want. There are ways to retry defined at the pipeline level which are more effective than having something HEC specific. See https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go as a starting point.

Reply:
Actually, I was referring to it before I started working on this PR. I am planning to add HEC ACK support, but based on the current implementation, the splunk_hec exporter is either preparing a batch or sending the batch to splunk. 

HEC Ack requires checking the status from the indexer asynchronously. It shouldn't block batching/sending next data. Also, what should be next step if HEC Ack returns non-ok status? 
There are 2 possibilities that I can think of:

1. send all data to splunk and wait until it receives ACK status for all the batches.
    - Here, it needs to keep track of each batch and its buffer state for creating subLog/subMetric/subTrace and combine all the failed batches
    - While it is waiting for ACK status, the worker is not doing any other tasks, so it wastes the time and delays the next pushData call
2. Segregate batching and sending (current implementation)
    - It requires to create an internal queue. But, since batching and sending are done in different go routines, what should it do when sending events fails?
      - It can retry batches from the internal queue only (so it doesn't need to report back to the exporter helper)
      - It can report back failed batches during the next push cycle (it increases complexity as it needs to create subLog/subMetric/subTrace, so it also requires to add plog data into queue)